src/comlink.ts

```diff
@@ -387,11 +403,36 @@ function isMessagePort(endpoint: Endpoint): endpoint is MessagePort {
   return endpoint.constructor.name === "MessagePort";
 }
 
+function isEndpoint(endpoint: object): endpoint is Endpoint {
+  return (
+    "postMessage" in endpoint &&
+    "addEventListener" in endpoint &&
+    "removeEventListener" in endpoint
+  );
+}
+
+function markLegacyPort(maybePort: unknown) {
+  if (
+    isObject(maybePort) &&
+    isEndpoint(maybePort) &&
+    isMessagePort(maybePort)
+  ) {
+    legacyEndpoints.add(maybePort);
+  }
+}
+
 function closeEndPoint(endpoint: Endpoint) {
   if (isMessagePort(endpoint)) endpoint.close();
 }
 
-export function wrap<T>(ep: Endpoint, target?: any): Remote<T> {
+export function wrap<T>(
+  ep: Endpoint,
+  target?: any,
+  legacy: boolean = false
```

Comment:
This is provided for cases where prior knowledge has not been obtained. For instance if instead of doing this:

```ts
a.foo(Comlink.transfer(port, [port])
```

The caller does:
```ts
a.foo(Comlink.transfer({ prop: port }, [port]))
```

In this case if inside foo we do:

```
foo({ prop: port }) {
  const remote = Comlink.wrap(port);
  // Oopsie this will break! It won't use the legacy format.
  remote.oopsie();
}
```

In this scenario because we don't traverse the object `{ prop: port }` we don't mark `port` as legacy. We could traverse them but this might have a huge overhead and so I opted for not doing it.

Instead the implementer can do:

```
foo({ prop: port }) {
  const remote = Comlink.wrap(port, undefined, true);
  // Ok fine this works :)
  remote.oopsie();
}
```

Reply:
If we want to add the traversal then only `markLegacyPort` needs to be updated